### DISCUSSION TITLE
Translate datatable batch actions

### DISCUSSION THREAD

**Discussion by reinisg** [carbon-design-system/carbon#14340] [Help] [answered]
Hello!
I am working with React Carbon components
I have packages:
"react": "^17.0.2",
"@carbon/react": "^1.30.0",

Im using datatable with batch actions from this example https://react.carbondesignsystem.com/?path=/story/components-datatable-batch-actions--default
and can not found a way how translate 'items selected' and 'cancel', [see in this image](https://i.ibb.co/4YJKz4Z/tablebatch.png) 

Is it possible to translate?

**Comment by tw15egan** [comment, marked as answer]
Definitely, you're going to want to pass in a custom function to `TableBatchActions` like so:

```jsx
<TableBatchActions translateWithId={translateWithId}>
```

and in that function, return whatever translation is needed for the two ids:


```js
function translateWithId(id) {
  if (id === 'carbon.table.batch.cancel') {
    return 'my cancel translation';
  }
  if (id === 'carbon.table.batch.items.selected') {
    return 'my items selected translation';
  }
}
```


More [translationIds](https://github.com/search?q=repo%3Acarbon-design-system%2Fcarbon-components-react+translationIds&type=code) and [translationKeys](https://github.com/search?q=repo%3Acarbon-design-system%2Fcarbon-components-react+translationKeys&type=code) across the repo. Please let us know if there are any strings that you are not able to translate. Hope that helps!

**Reply by reinisg** [reply]
Thank you a lot!! 🙌
